lib/datasets/ames-housing.rb

```diff
@@ -0,0 +1,131 @@
+                        :bsmt_unf_sf,
+                        :total_bsmt_sf,
+                        :heating,
+                        :heating_qc,
+                        :central_air,
+                        :electrical,
+                        :first_flr_sf,
+                        :second_flr_sf,
+                        :low_qual,
+                        :fin_sf,
+                        :gr_liv_area,
+                        :bsmt_full_bath,
+                        :bsmt_half_bath,      
+                        :full_bath,
+                        :half_bath,
+                        :bedroom_abv_gr,
+                        :kitchen_abv_gr,
+                        :kitchen_qual,
+                        :tot_rms_abv_grd,
+                        :functional,
+                        :fireplaces,
+                        :fireplace_qu,
+                        :garage_type,
+                        :garage_yr_blt,
+                        :garage_finish,
+                        :garage_cars,
+                        :garage_area,
+                        :garage_qual,
+                        :garage_cond,
+                        :paved_drive,
+                        :wood_deck_sf,
+                        :open_porch_sf,
+                        :enclosed_porch,
+                        :three_ssn_porch,
+                        :screen_porch,
+                        :pool_area,
+                        :pool_qc,
+                        :fence,
+                        :misc_feature,
+                        :misc_val,
+                        :mo_sold, 
+                        :yr_sold,
+                        :sale_type,
+                        :sale_condition,
+                        :sale_price)
+      
+
+    def initialize
+      super()
+      @metadata.id = "ames-housing"
+      @metadata.name = "Ames Housing"
+      @metadata.url = "http://jse.amstat.org/v19n3/decock/DataDocumentation.txt"
+      @metadata.licenses = ["Unknown"]
+      @metadata.description = <<-DESCRIPTION
+Data set contains information from the Ames Assessor’s Office
+used in computing assessed values for individual residential
+properties sold in Ames, IA from 2006 to 2010.
+De Cock, D.,
+"Ames, Iowa: Alternative to the Boston Housing Data as an
+End of Semester Regression Project",
+Journal of Statistics Education, 19(3) (2011) 1-15.
+Available from http://jse.amstat.org/v19n3/decock.pdf.
+      DESCRIPTION
+    end
+
+    def each
+      return to_enum(__method__) unless block_given?
+
+      open_data do |input|
+        input.each do |row|
+          next if row[0].nil?
+          record = Record.new(*row)
+          yield(record)
+        end
+      end
+    end
+
+    private
+    def open_data
+      data_path = cache_dir_path + "AmesHousing.txt"
+      data_url = "http://jse.amstat.org/v19n3/decock/AmesHousing.txt"
+      download(data_path, data_url)
+      CSV.open(data_path, converters: [:numeric]) do |csv|
```

Comment:
It's better that we preprocess the original data based on http://jse.amstat.org/v19n3/decock/DataDocumentation.txt .

For example, the "MS SubClass" column values can be converted to "1-STORY 1946 & NEWER ALL STYLES" from "020" and so on:

```text
MS SubClass (Nominal): Identifies the type of dwelling involved in the sale.	

       020	1-STORY 1946 & NEWER ALL STYLES
       030	1-STORY 1945 & OLDER
       040	1-STORY W/FINISHED ATTIC ALL AGES
       045	1-1/2 STORY - UNFINISHED ALL AGES
       050	1-1/2 STORY FINISHED ALL AGES
       060	2-STORY 1946 & NEWER
       070	2-STORY 1945 & OLDER
       075	2-1/2 STORY ALL AGES
       080	SPLIT OR MULTI-LEVEL
       085	SPLIT FOYER
       090	DUPLEX - ALL STYLES AND AGES
       120	1-STORY PUD (Planned Unit Development) - 1946 & NEWER
       150	1-1/2 STORY PUD - ALL AGES
       160	2-STORY PUD - 1946 & NEWER
       180	PUD - MULTILEVEL - INCL SPLIT LEV/FOYER
       190	2 FAMILY CONVERSION - ALL STYLES AND AGES
```